src/mongo/tree/MongoAccountTreeItem.ts

```diff
@@ -73,7 +73,14 @@ export class MongoAccountTreeItem extends AzureParentTreeItem<IMongoTreeRoot> {
             return databases
                 .filter((database: IDatabaseInfo) => !(database.name && database.name.toLowerCase() === "admin" && database.empty)) // Filter out the 'admin' database if it's empty
                 .map(database => new MongoDatabaseTreeItem(this, database.name, this.connectionString));
-        } finally {
+        } catch (error) {
+            let message = parseError(error).message;
+            if (this._root.isEmulator && message.includes("ECONNREFUSED")) {
+                error.message = "Unable to reach emulator. Please ensure it is started and writing to the appropriate port. Then try again.";
```

Comment:
Still not really happy with this approach, although I understand the logic. If you're going to use it, at least add the original message to the end, thanks.

Ideally we'd have a common util to do this properly.

Reply:
Do you prefer this PR to add a common util of the form: 
```
prefixErrorWithMessage(error, message) ..

appendErrorWithMessage(error, message)...
```

Reply:
Would love to see that, yeah.  How about nestError(errorish: unknown, mainMessage: string)